hamilton/plugins/pandas_extensions.py

```diff
@@ -4,24 +4,28 @@
 from collections.abc import Hashable
 from io import BufferedReader, BytesIO, StringIO
 from pathlib import Path
+from sqlite3 import Connection
 from typing import Any, Callable, Collection, Dict, Iterator, List, Optional, Tuple, Type, Union
 
 try:
-    import pandas as pd
+    from collections.abc import Iterable, Sequence
 except ImportError:
-    raise NotImplementedError("Pandas is not installed.")
+    from collections import Iterable, Sequence
 
 try:
     from typing import Literal
 except ImportError:
     from typing_extensions import Literal
 
 try:
-    from collections.abc import Iterable, Sequence
+    import pandas_gbq
 except ImportError:
-    from collections import Sequence, Iterable
+    raise NotImplementedError("pandas_gbq is not installed.")
```

Comment:
let's place this check in the dataclass itself. Ideally in the "constructor" portion, i.e. when someone goes to create the materializer we should check the import. E.g. as a post init thing  https://docs.python.org/3/library/dataclasses.html#post-init-processing.

Reply:
We can then do the "import" within the `load_data` and `save_data` methods to remove the need to do it here.